src/modules/CC1101/CC1101.cpp

```diff
@@ -298,31 +294,66 @@ int16_t CC1101::readData(uint8_t* data, size_t len) {
     SPIreadRegister(CC1101_REG_FIFO);
   }
 
-  // read packet data
-  SPIreadRegisterBurst(CC1101_REG_FIFO, length, data);
+  uint8_t bytesInFIFO = SPIgetRegValue(CC1101_REG_RXBYTES, 6, 0);
+  uint16_t readBytes = 0;
+  unsigned long lastPop = millis();
+
+  // keep reading from FIFO until we get all the packet.
+  while (readBytes < length) {
+    if (bytesInFIFO == 0) {
+      if (millis() - lastPop > 5) {
```

Comment:
If radio is in fixed mode and PKTLEN is set to a value a way higher that the actual received packet, the modem will just stop feeding the FIFO.